skipchain/api.go

```diff
@@ -177,10 +177,33 @@ func (c *Client) CreateRootControl(elRoot, elControl *onet.Roster,
 // 'latest' skipblock and the id (=hash) of the latest skipblock.
 func (c *Client) GetUpdateChain(roster *onet.Roster, latest SkipBlockID) (reply *GetUpdateChainReply, err error) {
 	reply = &GetUpdateChainReply{}
-	r := roster.RandomServerIdentity()
-	err = c.SendProtobuf(r,
-		&GetUpdateChain{latest}, reply)
-	return
+	for {
+		r2 := &GetUpdateChainReply{}
+		r := roster.RandomServerIdentity()
+		err = c.SendProtobuf(r, &GetUpdateChain{LatestID: latest}, r2)
+		if err != nil {
+			return
```

Comment:
I think you should not return here but try another server from the list. Either iterate through the roster, or randomly iterate through it. But some of the servers might be down.